algorithms/linfa-ica/benches/fast_ica.rs

```diff
@@ -0,0 +1,60 @@
+use criterion::{criterion_group, criterion_main, BenchmarkId, Criterion};
+use linfa::{dataset::DatasetBase, traits::Fit};
+use linfa_ica::fast_ica::{FastIca, GFunc};
+use ndarray::{array, concatenate};
+use ndarray::{Array, Array2, Axis};
+use ndarray_rand::{rand::SeedableRng, rand_distr::Uniform, RandomExt};
+use rand_xoshiro::Xoshiro256Plus;
+
+fn perform_ica(size: usize) -> () {
+    let sources_mixed = create_data(size);
+
+    let ica = FastIca::params().gfunc(GFunc::Logcosh(1.0));
+
+    let ica = ica.fit(&DatasetBase::from(sources_mixed.view()));
+}
+
+fn create_data(nsamples: usize) -> Array2<f64> {
+    // Creating a sine wave signal
+    let source1 = Array::linspace(0., 8., nsamples).mapv(|x| (2f64 * x).sin());
+
+    // Creating a sawtooth signal
+    let source2 = Array::linspace(0., 8., nsamples).mapv(|x| {
+        let tmp = (4f64 * x).sin();
+        if tmp > 0. {
+            return 1.;
+        }
+        -1.
+    });
+
+    // Column concatenating both the signals
+    let mut sources_original = concatenate![
+        Axis(1),
+        source1.insert_axis(Axis(1)),
+        source2.insert_axis(Axis(1))
+    ];
+
+    // Adding noise to the signals
+    let mut rng = Xoshiro256Plus::seed_from_u64(42);
+    sources_original +=
+        &Array::random_using((nsamples, 2), Uniform::new(0.0, 1.0), &mut rng).mapv(|x| x * 0.2);
+
+    // Mixing the two signals
+    let mixing = array![[1., 1.], [0.5, 2.]];
+    let sources_mixed = sources_original.dot(&mixing.t());
+
+    sources_mixed
+}
+
+fn bench(c: &mut Criterion) {
+    let mut group = c.benchmark_group("fast_ica_bench");
+    for size in [1_000, 10_000, 100_000].iter() {
```

Comment:
How long does 100_000 samples take? It always takes too long on my machine, but maybe ICA is a lot faster than other algorithms.

Reply:
```
fast_ica_bench/fast-ica-{}/100000
                        time:   [15.202 ms 15.575 ms 15.967 ms]
Found 5 outliers among 100 measurements (5.00%)
  5 (5.00%) high mild

```

Reply:
Nice